docs/plugins/ingest.asciidoc

```diff
@@ -101,9 +87,7 @@ Throws error when the field is not an array.
 --------------------------------------------------
 {
   "join": {
-    "fields": {
-      "joined_array_field": "other_array_field"
-    }
+    "joined_array_field": "-"
```

Comment:
+10 on more explicit configuration... here's the thing. the key:value approach, not only that it make little sense semantically, it's not extensible in a consistent manner. If at some point in the future we'll need to add another setting to it, we'll end up with something like:

```json
"join" : {
  "joined_array_field": {
    "separator" : "-",
    "extra-setting" : "setting value"
  }
}
```
and now you have (again) two ways of configuring something.. vs... if we keep it extensible from day one:

```json
"join" : {
  "field" : "joined_array_field",
  "separator" : "-",
  "extra-setting" : "setting value"
}
```